src/hotspot/share/code/SCCache.cpp

```diff
@@ -580,10 +589,28 @@ SCCache::SCCache(const char* cache_path, int fd, uint load_size) {
   _table = new SCAddressTable();
 }
 
-void SCCache::init_table() {
+void SCCache::init_extrs_table() {
+  SCCache* cache = SCCache::cache();
+  if (cache != nullptr && cache->_table != nullptr) {
+    cache->_table->init_extrs();
```

Comment:
This is repetitive checks pattern. I suggest to replace it with check of call to:
```
  SCAddressTable* SCCache::addr_table() const {
    return is_on() && (cache()->_table != nullptr) ? cache()->_table : nullptr;
  }
```

Reply:
Done